### PR TITLE
[libssh2,libgit2,vcpkg-ci-libgit2] Fix, test, cleanup

### LINKED ISSUE #32433: [libgit2] Build error
Package: libgit2[core,openssl,pcre2,ssh]:x64-windows -> 1.6.4

**Host Environment**

- Host: x64-windows
- Compiler: MSVC 19.36.32535.0
-    vcpkg-tool version: 2023-06-15-8c254a5fb6c503615834fc66bd0717664a339a2e
    vcpkg-scripts version: 228987a6f 2023-07-05 (2 days ago)

**To Reproduce**

`vcpkg install libgit2[core,openssl,pcre2,ssh]`

**Failure logs**

```
-- Using cached libgit2-libgit2-v1.6.4.tar.gz.
-- Extracting source C:/ToolChain/vcpkg/downloads/libgit2-libgit2-v1.6.4.tar.gz
-- Applying patch fix-configcmake.patch
-- Using source at C:/ToolChain/vcpkg/buildtrees/libgit2/src/v1.6.4-6e3df5985c.clean
-- Found external ninja('1.11.0').
-- Configuring x64-windows
CMake Error at scripts/cmake/vcpkg_execute_required_process.cmake:112 (message):
    Command failed: "C:/Program Files (x86)/Microsoft Visual Studio/2022/BuildTools/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" -v
    Working Directory: C:/ToolChain/vcpkg/buildtrees/libgit2/x64-windows-rel/vcpkg-parallel-configure
    Error code: 1
    See logs for more information:
      C:\ToolChain\vcpkg\buildtrees\libgit2\config-x64-windows-dbg-CMakeCache.txt.log
      C:\ToolChain\vcpkg\buildtrees\libgit2\config-x64-windows-rel-CMakeCache.txt.log
      C:\ToolChain\vcpkg\buildtrees\libgit2\config-x64-windows-out.log

Call Stack (most recent call first):
  installed/x64-windows/share/vcpkg-cmake/vcpkg_cmake_configure.cmake:252 (vcpkg_execute_required_process)
  ports/libgit2/portfile.cmake:62 (vcpkg_cmake_configure)
  scripts/ports.cmake:147 (include)
```

### PR DESCRIPTION
**Pull request by dg0yt**
# libssh2

Fixes pc file for windows.
Enable building windows without openssl.

# libgit2

Generally leverages pkg-config for dependencies:
Fixes https://github.com/microsoft/vcpkg/issues/32433 (ssh) by using libssh2 pc files.
Fixes pcre2 debug lib name mismatch.

Rectifies exported unofficial cmake config: 
Provide canonical `unofficial-<port>` package/`unofficial::<port>::libgit2` target.
Drops pointless `find_dependency`.
Adds include dir property.

Adds usage hint, for pkg-config and cmake.

Fixes android builds (https://github.com/libgit2/libgit2/issues/6574).

Add `tools` feature.

Fix mingw.

Revise copyright notice.

# vcpkg-ci-libgit2

New test port, activates ssh and tools features.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [x] SHA512s are updated for each updated download
- [x] The "supports" clause reflects platforms that may be fixed by this new version
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/libgit2/unofficial-libgit2-config.cmake [hunk 1]
```diff
@@ -0,0 +1,3 @@
+include("${CMAKE_CURRENT_LIST_DIR}/unofficial-libgit2-target.cmake")
```
Comment (LilyWangLL) [inline ports/libgit2/unofficial-libgit2-config.cmake:1]:
```suggestion
include("${CMAKE_CURRENT_LIST_DIR}/unofficial-libgit2-targets.cmake")
```

Comment (LilyWangLL) [reply ports/libgit2/unofficial-libgit2-config.cmake:1]: And `unofficial-libgit2-targets.cmake` doesn't have include properties.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
All features of `libssh2` and feature `libgit2[tools]` passed on the following triplets:
```
x86-windows
x64-windows
x64-windows-static
```
`Usage` of `libssh2` passed on x64-windows.

**Comment by LilyWangLL** [comment]
Note: I will be converting your PR to draft status. When you respond, please revert to "ready for review". That way, I can be aware that you've responded since you can't modify the tags.

File: ports/libgit2/unofficial-libgit2-config.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/libgit2/unofficial-libgit2-config.cmake]: Thank your for catching the typo.

> And `unofficial-libgit2-targets.cmake` doesn't have include properties.

Not a regression, but worth fixing indeed. Done.

**Review by LilyWangLL — APPROVED** [review]

**Comment by BillyONeal** [comment]
Thanks!

**Pull request merged**